src/helpers/discord.ts

```diff
@@ -41,3 +42,6 @@ export const fetchReactionMembers = (
     .then((users) =>
       Promise.all(users.map((user) => guild.members.fetch(user.id))),
     );
+
+export const fetchMessagesByUser = (thread: ThreadChannel, userID: string) =>
+  thread.messages.cache.filter((message) => message.author.id === userID);
```

Comment:
This relies on the messages being cached, which won't be true for threads that were created before the bot restarted. Is it feasible to make this something like, `thread.messages.fetch()`? (I don't have a dev computer handy or I'd check, sorry)